Docs/source/usage/parameters.rst

```diff
@@ -1469,6 +1469,18 @@ Boundary conditions
     Whether to damp current in PML. Can only be used if particles are propagated in PML,
     i.e. if `warpx.pml_has_particles = 1`.
 
+* ``warpx.do_pml_dive_cleaning`` (`bool`; default: 1)
+    Whether to use divergence cleaning for E in the PML region.
+    The value must match ``warpx.do_pml_divb_cleaning`` (either both false or both true).
+    This option seems to be necessary in order to avoid strong Nyquist instabilities in 3D simulations with open boundary conditions and PML in all directions. 2D simulations and 3D simulations with open boundary conditions and PML only in one direction might run well even without divergence cleaning.
+    This option is implemented only for the PSATD solver.
+
+* ``warpx.do_pml_divb_cleaning`` (`bool`; default: 1)
+    Whether to use divergence cleaning for B in the PML region.
+    The value must match ``warpx.do_pml_dive_cleaning`` (either both false or both true).
+    This option seems to be necessary in order to avoid strong Nyquist instabilities in 3D simulations with open boundary conditions and PML in all directions. 2D simulations and 3D simulations with open boundary conditions and PML only in one direction might run well even without divergence cleaning.
+    This option is implemented only for the PSATD solver.
+
```

Comment:
This is a good point that we should probably discuss "in person", in one of our developers meetings.

I think that, from a theoretical point of view, it would make sense to keep the two options separate, because there could be scenarios where it is sufficient to activate only one of the two cleaning methods. As of today, we have not encountered such scenarios yet.

The reason why I currently require that they are either both false or both true is that I am not sure that the equations that I derived and implemented, which included both F and G from the beginning, happen to reduce naturally to the correct equations that would be derived starting off with just one of the two fields in the first place (F or G).

Since I have not checked this properly and since we do not have a test case for this, I decided to implement only the two cases above.

What I can do, certainly, is improve the abort message that I added for the case when `do_pml_dive_cleaning` differs from `do_pml_divb_cleaning`: I could clearly state that this case has not been implemented yet (rather than saying that "they should be equal"). If we then realize that the current code can be trivially modified to cover that case too, because the current equations extend naturally, we can do that later on once we have a test case for it.

I will modify the abort message as outlined above in the meantime and will be looking forward to hearing your comments on this.

Reply:
I did this in https://github.com/ECP-WarpX/WarpX/pull/1600/commits/bee7ffbffb2b4a168b34e589956c7d3a38a70232. The new abort message now looks like the following:
```
amrex::Abort::0::
warpx.do_pml_dive_cleaning = 1 and warpx.do_pml_divb_cleaning = 0:
this case is not implemented yet, please set both parameters to the same value !!!
```

Reply:
Great! I think that this is a good solution...